### DISCUSSION TITLE
How to bump bitnami/spark both version and "r"?

### DISCUSSION THREAD

**Discussion by hongbo-miao** [renovatebot/renovate#26460] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

github.com

### What is your question?

Renovate opens [this pull request](https://github.com/hongbo-miao/hongbomiao.com/pull/13508) by upgrading Docker image **bitnami/spark** version from

1. `3.4.1-debian-11-r1`
2. `3.4.2-debian-11-r1`

Basically Renovate thinks it has to end with `-debian-11-r1` currently.

However, I am hoping the middle `-debian-11` fixed, Renovate could upgrade both version and "r" from

1. `3.4.1-debian-11-r1`
2. `3.4.2-debian-11-r0` (Renovate missed)
3. `3.4.2-debian-11-r1`
4. ...
5. `3.5.0-debian-11-r0` (Renovate missed)
6. `3.5.0-debian-11-r1` (Renovate missed)
7. ...
8. `3.5.0-debian-11-r17` (Renovate missed, latest version as of today)

Is there a way? Thanks! ☺️

**Comment by viceice** [comment, marked as answer]
I'm using a custom regex versioning

https://github.com/visualon/renovate-config/blob/acf2a87b1f6600aabaaea5bc7e22c9bfa922f279/shared.json#L133

**Reply by hongbo-miao** [reply]
Thanks @viceice ! Just confirm this will help upgrade both version and "r", but not including the middle `debian-11`, and I need add to my `packageRules`, right?
If it is, should this be default behavior for bitnami Docker packages? (probably most users expect?).

**Reply by viceice** [reply]
it will bump to `debian-12` when available.

and yes, you need a package rule which should be limited to bitnami images.

**Reply by hongbo-miao** [reply]
Oh I was thinking bumping to `debian-12` also makes sense. Thanks! Kind of feeling it should be default Renovate behavior without user adding `packageRules` manually though.

Ended with

```json
{
  ...
  "packageRules": [
    {
      "matchPackageNames": [
        "docker.io/bitnami/spark"
      ],
      "versioning": "regex:^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(:?-(?<compatibility>.+)(?<build>\\d+)-r(?<revision>\\d+))?$"
    }
  ]
}
```

And works well now!
